api/src/database/migrations/20240311A-deprecate-webhooks.ts

```diff
@@ -0,0 +1,145 @@
+import { randomUUID } from '@directus/random';
+import { parseJSON, toArray } from '@directus/utils';
+import type { Knex } from 'knex';
+import type { Webhook } from '../../types/webhooks.js';
+
+// To avoid typos
+const TABLE_WEBHOOKS = 'directus_webhooks';
+const TABLE_FLOWS = 'directus_flows';
+const TABLE_OPERATIONS = 'directus_operations';
+const NEW_COLUMN_WAS_ACTIVE = 'was_active_before_deprecation';
+const NEW_COLUMN_FLOW = 'migrated_flow';
+
+/**
+ * 0. Identify and persist which webhooks were active before deprecation
+ * 1. Migrate existing webhooks over to identically behaving Flows
+ * 2. Disable existing webhooks
+ * 3. Dont drop webhooks yet
+ */
+export async function up(knex: Knex): Promise<void> {
+	// 0. Identify and persist which webhooks were active before deprecation
+	await knex.schema.alterTable(TABLE_WEBHOOKS, (table) => {
+		table.boolean(NEW_COLUMN_WAS_ACTIVE).notNullable().defaultTo(false);
+		table.uuid(NEW_COLUMN_FLOW).nullable();
+		table.foreign(NEW_COLUMN_FLOW).references(`${TABLE_FLOWS}.id`).onDelete('SET NULL');
+	});
+
+	await knex(TABLE_WEBHOOKS)
+		.update({ [NEW_COLUMN_WAS_ACTIVE]: true })
+		.where({ status: 'active' });
+
+	const webhooks: Webhook[] = await knex.select('*').from(TABLE_WEBHOOKS);
+
+	// 1. Migrate existing webhooks over to identically behaving Flows
+	await knex.transaction(async (trx) => {
+		for (const webhook of webhooks) {
+			const flowId = randomUUID();
+			const operationIdRunScript = randomUUID();
+			const operationIdWebhook = randomUUID();
+
+			const newFlow = {
+				id: flowId,
+				name: webhook.name,
+				icon: 'webhook',
+				// color: null,
+				description:
+					'Auto-generated by Directus Migration\n\nWebhooks were deprecated and have been moved into Flows for you automatically.',
+				status: webhook.status, // Only activate already active webhooks!
+				trigger: 'event',
+				// accountability: "all",
+				options: {
+					type: 'action',
+					scope: toArray(webhook.actions)
+						.filter((action) => action.trim() !== '')
+						.map((scope) => `items.${scope}`),
+					collections: toArray(webhook.collections),
```

Comment:
Saw that collections can also be empty 🙈 
```suggestion
					collections: toArray(webhook.collections).filter((collection) => collection.trim() !== ''),
```